src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -2557,16 +2557,17 @@ bool G1CollectedHeap::print_location(outputStream* st, void* addr) const {
 
 G1HeapSummary G1CollectedHeap::create_g1_heap_summary() {
 
-  size_t eden_used_bytes = _eden.used_bytes();
-  size_t survivor_used_bytes = _survivor.used_bytes();
+  size_t eden_used_bytes = _monitoring_support->eden_space_used();
+  size_t survivor_used_bytes = _monitoring_support->survivor_space_used();
+  size_t old_used_bytes = _monitoring_support->old_gen_used();
```

Comment:
I think it's best to include `gen` in the name (across the whole PR), just to avoid the confusion of Old-region-type vs old-generation.